src/pubtools/_marketplacesvm/tasks/delete/command.py

```diff
@@ -0,0 +1,271 @@
+                log.info("Would have updated image %s in rhsm", push_item.image_id)
+                return
+
+            log.info("Attempting to update the existing image %s in rhsm", push_item.image_id)
+            out = self.rhsm_client.aws_update_image(**image_meta)
+            resp = out.result()
+            if not resp.ok:
+                log.error("Failed updating image %s", push_item.image_id)
+                resp.raise_for_status()
+
+            log.info("Existing image %s succesfully updated in rhsm", push_item.image_id)
+
+    def collect_push_result(self, results: List[AmiPushItem]) -> Dict[str, Any]:
+        """
+        Collect the push results and sends its json to the collector.
+
+        Args:
+            results
+                The list of dictionaries containing the result data for the push collector.
+
+        Returns:
+            The result of push collector `attach_file` call.
+        """
+
+        def convert(obj):
+            if isinstance(obj, (datetime.datetime, datetime.date)):
+                return obj.strftime("%Y%m%d")
+
+        push_items = []
+        mod_result = []
+        for result in results:
+            res_dict = asdict(result)
+            # dict can't be modified during iteration.
+            # so iterate over list of keys.
+            for key in list(res_dict):
+                if res_dict[key] is None:
+                    del res_dict[key]
+            mod_result.append(res_dict)
+            push_items.append(result)
+
+        metadata = json.dumps(mod_result, default=convert, indent=2, sort_keys=True)
+        self.collector.update_push_items(push_items).result()
+        return self.collector.attach_file("clouds.json", metadata).result()
+
+    def _convert_provider_name(self, provider_name: str) -> List[str]:
+        # Since we're only focused on AWS these are the accounts we're dealing with.
+        # AmiProduct is a list as there's no way of telling which account the image
+        # is on until we try to delete it.
+        # Eventually we might want to add to pushsource a provider name field.
+        accounts = {
+            "AWS": ["aws-us-storage"],
+            "AGOV": ["aws-us-gov-storage"],
+            "ACN": ["aws-china-storage"],
+            "AmiProduct": ["aws-na", "aws-emea"],
+        }
+        return accounts[provider_name]
+
+    def _delete(
+        self,
+        push_item: VMIPushItem,
+        **kwargs,
+    ) -> VMIPushItem:
+        builds = self.args.builds
+        dry_run = self.args.dry_run
+        keep_snapshots = self.args.keep_snapshots
+        marketplaces = self._convert_provider_name(push_item.marketplace_entity_type)
+        for marketplace in marketplaces:
+            if dry_run:
+                log.info("Would have deleted: %s in build %s", push_item.image_id, push_item.build)
+                self._SKIPPED = True
+                pi = evolve(push_item, state=State.SKIPPED)
+            elif push_item.build in builds:
```

Comment:
Please replace this by `continue` then you save up indentation space

```suggestion
                pi = evolve(push_item, state=State.SKIPPED)
                continue
            push_item.build in builds:
```

Reply:
This is a check to see if the build is in builds marked to be deleted. Since we can do a push item and specify only part of it is to be deleted.

Reply:
Ah, ok, got it!

You have an else condition at the end as well, marking this as resolved

Reply:
oh wait actually now that you mention it... this isn't quite right

Reply:
this is cleaned up a bit